app/src/main/java/com/example/bigowlapp/viewModel/SignUpViewModel.java

```diff
@@ -19,26 +19,17 @@ public SignUpViewModel() {
     }
 
     // TODO: Create a default group upon user creation
-    public Task<Boolean> createUser(String email, String password, String phoneNumber, String name) {
-        return authRepository.signUpUser(email, password, phoneNumber, nameExtractor(name)[0],
-                nameExtractor(name)[1])
+    public Task<Boolean> createUser(String email, String password, String phoneNumber, String firstName, String lastName) {
+        return authRepository.signUpUser(email, password, phoneNumber, firstName, lastName)
                 .addOnSuccessListener(isSuccess -> {
                     Group group = new Group();
                     group.setMonitoringUserId(authRepository.getCurrentUser().getUid());
-                    group.setName(name + "'s group " + "#" + randomStringGenerator());
+                    group.setName(firstName + "'s group " + "#" + randomStringGenerator());
                     groupRepository.addDocument(group);
                 });
 
     }
 
-    public String nameParcer(String firstName, String lastName) {
-        return firstName + " " + lastName;
-    }
-
```

Comment:
For this, you should add back that nameParcer fuction (although rename it like `getFullName()`), and set group name using that.

We should keep the original behavior of that code in line 27/28 where the groups name used the full name